src/core/agent/services/ipexCommunicationService.test.ts

```diff
@@ -328,6 +349,272 @@ describe("Ipex communication service of agent", () => {
+      ],
+    });
+    createExchangeMessageMock.mockResolvedValueOnce([
+      multisigExnIpexGrantSerder,
+      multisigExnIpexGrantSig,
+      multisigExnIpexGrantEnd,
+    ]);
+
+    await ipexCommunicationService.multisigAdmit(
+      multisigId,
+      notificationSaid,
+      schemaSaids
+    );
+    expect(ipexAdmitMock).toBeCalledTimes(1);
+    expect(createExchangeMessageMock).toBeCalledTimes(1);
+    expect(submitAdmitMock).toBeCalledTimes(1);
+  });
+
+  test("Can agree to admit a credential with a multi-sig identifier", async () => {
+    const multisigId = gHab.name;
+    const notificationSaid = "ELykd_2bX6yvuVEgLQqnCgZ7QLdxpUBze-RzHVwfCUfW";
+    const schemaSaids = ["schemaSaid"];
+    const multisigExn = {
+      v: "KERI10JSON000111_",
+      t: "exn",
+      d: "EO6rXPzJVLNEOjs3puI5Kn4L2UsiB-iJJJKpXi26F73X",
+      i: "EHQDKkV40qP65N8yHaOJFlVS1CUvsYTvGlHPfcy2tFUb",
+      p: "EKWG4i9hT8vjwRPHsW7vqWrPq0utZHVgdu24fAf0j2Cb",
+      dt: "2024-08-02T07:06:24.884000+00:00",
+      r: "/ipex/admit",
+      q: {},
+      a: { m: "" },
+      e: {},
+    };
+
+    const ked = {
+      v: "KERI10JSON000111_",
+      t: "exn",
+      d: "EKm404jyX0iquIOu0BtZ6xR04opEQYoClKeSTuuS4fwn",
+      i: "EL3BEUfwxS1_mCWqKUrH5nGPkikoiHskhVhGenV2lcAZ",
+      p: "EBXi4JFZqjsKMzaMAz-gJWxJj992R988JcdN8EfzL4Po",
+      dt: "2024-08-02T07:11:59.510000+00:00",
+      r: "/ipex/admit",
+      q: {},
+      a: { m: "" },
+      e: {},
+    };
+
+    const mockSaider = [{} as Saider, ked] as [Saider, Dict<any>];
+
+    connections.resolveOobi.mockResolvedValueOnce({
+      name: "oobi.AM3es3rJ201QzbzYuclUipYzgzysegLeQsjRqykNrmwC",
+      metadata: {
+        oobi: "testOobi",
+      },
+      done: true,
+      error: null,
+      response: {},
+      alias: "c5dd639c-d875-4f9f-97e5-ed5c5fdbbeb1",
+    });
+    const exchangeMock = {
+      exn: {
+        i: "ELWFo-DV4GujnvcwwIbzTzjc-nIf0ijv6W1ecajvQYBY",
+        d: "EO65SZOen5Qm26gYeAZZ_J_p8_Uy_6jB3cUpv0DzgDA4",
+      },
+    };
+    getExchangeMock.mockResolvedValueOnce(exchangeMock);
+    identifiersMemberMock = jest.fn().mockResolvedValueOnce({
+      signing: [
+        { ends: { agent: { [memberMetadataRecord.id]: "" } }, aid: "aid" },
+      ],
+    });
+    identifierStorage.getIdentifierMetadata = jest.fn().mockResolvedValueOnce(
+      new IdentifierMetadataRecord({
+        id: "aidHere",
+        displayName: "Identifier 2",
+        signifyName: "uuid-here",
+        createdAt: now,
+        theme: 0,
+        groupMetadata: {
+          groupId: "group-id",
+          groupInitiator: true,
+          groupCreated: true,
+        },
+      })
+    );
+    identifiersGetMock = jest
+      .fn()
+      .mockResolvedValueOnce(gHab)
+      .mockResolvedValueOnce(mHab);
+
+    jest.spyOn(Saider, "saidify").mockReturnValueOnce(mockSaider);
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    jest.spyOn(require("signify-ts"), "Serder").mockImplementationOnce(() => {
```

Comment:
Hmm, to me we shouldn't need to mock primitive things in Signify like `saidify` and `Serder` - we can just pass it data that is valid, and it'll return the correct thing. Why do we need this spy?

Reply:
I mock saidify and Serder  because i cant mock return value of multiSigAdmit

Reply:
You're not meant to mock `multiSigAdmit` because it's now part of this file. We test the public functions in this file that are used by the UI or other services, and those public functions might call private functions like `multiSigAdmit` within the same file.

We mock external things, yes, but I don't think we need to mock primitive things in Signify. We should mock the Signify functions that query KERIA - so we are effectively mocking what is coming from KERIA

Reply:
It works to mock `saidify` or `Serder` but to me it's a sign that there's something else wrong, and there should be a simpler way to do this